extensions/quick-access/package.json

```diff
@@ -0,0 +1,125 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "quick-access",
+  "title": "Quick Access",
+  "description": "Quick copy, open, reveal the latest files in pinned Finder directories.",
+  "icon": "extension-icon.png",
+  "author": "koinzhang",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "pin-directory",
+      "title": "Pin Directory",
+      "icon": "pin-directory.png",
+      "subtitle": "Quick Access",
+      "description": "Pin directories that can be accessed quickly.",
+      "mode": "no-view"
+    },
+    {
+      "name": "access-directory",
+      "title": "Access Directory",
+      "icon": "access-directory.png",
+      "subtitle": "Quick Access",
+      "description": "Quick copy, open, reveal the latest files in pinned Finder directories.",
+      "mode": "view",
+      "preferences": [
+        {
+          "name": "autoCopyLatestFile",
+          "type": "checkbox",
+          "required": true,
+          "title": "Advanced Preference",
+          "label": "Auto Copy Latest File",
+          "description": "Automatically copy the latest files in the topmost directory.",
+          "default": true
+        },
+        {
+          "name": "showDeleteAction",
+          "type": "checkbox",
+          "required": true,
+          "label": "Show Delete File Action",
+          "description": "Show permanent file deletion action, please check carefully.",
+          "default": true
+        },
+        {
+          "name": "fileShowNumber",
+          "type": "dropdown",
+          "required": true,
+          "title": "Number of files shown",
+          "description": "Number of files displayed per directory.",
+          "default": "3",
```

Comment:
Is this preference really necessary? If you keep it, I would suggest to use "All" as default option and remove the `required` option.

Reply:
Here I think it is necessary.
First let me explain the inspiration for this extension. One of the partners in the community wanted to create an extension: Copy Download Latest File, which is to copy the latest file in the download directory.

I was inspired by the idea that when we download, take a screenshot, or save a file to some common folder, like the download folder or the screenshot folder, and we want to copy or open the file immediately, we can quickly get the latest file with the help of this extension (so the files in the pinned directory are sorted by modification time).
If I only want to get the latest few files, this preference is necessary, especially in a small view window like raycast, if  show all the files, then the files in a directory may fill up the whole window, and it becomes very troublesome to find the files.

Many download files or screenshots from a long time ago are actually invalid, so it is very necessary to be able to display only the latest few files.

The functionality of this extension is similar to Raycast's newly updated Search Screnshot, except that this extension will display all types of files.

<img width="1000" alt="Quick Access" src="https://user-images.githubusercontent.com/36128970/164712178-a45331db-feac-4ef1-8cfd-88e8319ad6c4.png">
<img width="1000" alt="Search Screenshot" src="https://user-images.githubusercontent.com/36128970/164712196-570b6546-b3f0-40f1-9efb-094f7967b8d5.png">

Reply:
Wouldn't sorting the files by recently modified/added solve this problem? Than the latest file is always at the top. It's just confusing that I don't see all my downloads by default. I didn't expect that and thought the extension is broken.

Reply:
Indeed it may confuse the user.
So here I will change the default to all, but the preferences will remain so that there can be more options.

Reply:
Sounds good. I wouldn't make it required. Then the user can start using the extension and can optimize in the preferences later.

Generally, required preferences make mostly sense for things that need to be set up to make the extension work. This pref doesn't seem to be the case. So it's better to keep it non-required.

Reply:
In fact, I prefer to set the require to True for some preferences.
Because most of the time, users don't know that your extension has a preference option and rarely go to the preferences page to check it.